src/client/localplayer.cpp

```diff
@@ -492,6 +497,12 @@ void LocalPlayer::move(f32 dtime, Environment *env, f32 pos_max_d,
 	// Prevent sliding on the ground when jump speed is 0
 	m_can_jump = m_can_jump && jumpspeed != 0.0f;
 
+	// Prevent jump if in node/object
+	if (m_can_jump) {
+		if (collisionInCollision(env, m_client, m_collisionbox, initial_position))
+			m_can_jump = false;
+	}
```

Comment:
Preventing jumping because the player is inside a node is quite problematic.

If you repeatedly jump up while placing nodes below yourself (["pillaring"](https://minecraft.wiki/w/Tutorials/Pillar_jumping)), you are often partially inside a node for a short time (see #14068). With this change, you often get stuck while pillaring.

Also, other players can place nodes inside you on a multiplayer server. If you can't escape by sideways movement, the only way to escape is jumping, which is made impossible by this change.

Reply:
Understand.
Bug prevention does not work without it. But I added a sneak condition to that check.